internal/provider/resources/install/gcp/gcp.go

```diff
@@ -0,0 +1,138 @@
+package installgcp
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/p0-security/terraform-provider-p0/internal"
+	installresources "github.com/p0-security/terraform-provider-p0/internal/provider/resources/install"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &Gcp{}
+var _ resource.ResourceWithImportState = &Gcp{}
+var _ resource.ResourceWithConfigure = &Gcp{}
+
+func NewGcp() resource.Resource {
+	return &Gcp{}
+}
+
+type Gcp struct {
+	installer *installresources.RootInstall
+}
+
+type gcpModel struct {
+	OrganizationId      basetypes.StringValue `tfsdk:"organization_id"`
+	ServiceAccountEmail basetypes.StringValue `tfsdk:"service_account_email"`
+}
+
+type gcpApi struct {
+	Config struct {
+		Root struct {
+			Singleton struct {
+				OrganizationId      string  `json:"organizationId"`
+				ServiceAccountEmail *string `json:"serviceAccountEmail"`
+			} `json:"_"`
+		} `json:"root"`
+	} `json:"config"`
+}
+
+func (r *Gcp) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_gcp"
+}
+
+func (r *Gcp) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: `A Google Cloud installation.`,
+		Attributes: map[string]schema.Attribute{
+			"organization_id": schema.StringAttribute{
```

Comment:
Why isn't this a NumberAttribute?

https://developer.hashicorp.com/terraform/plugin/framework/handling-data/attributes/number

Reply:
Just because these identifiers are strings in our backend. We could presumably translate in the CRUD methods, but that's an extra step (presumably we'd still want a validator to ensure the number is positive). Also, I'm not 100% confident that org IDs are only numeric; changing this later would require our customers to rev their tf files.

Reply:
Interestingly, they also appear to be strings in https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/organization#org_id